etna/transforms/timestamp/holiday.py

```diff
@@ -83,27 +139,45 @@ def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         -------
         :
             pd.DataFrame with added holidays
+
+        Raises
+        ------
+        ValueError:
+            if the frequency is greater than daily and this is a ``binary`` or ``categorical`` mode
+        ValueError:
+            if the frequency is not weekly, monthly, quarterly or yearly data and this is ``days_count`` mode
         """
-        if (df.index[1] - df.index[0]) > datetime.timedelta(days=1):
-            raise ValueError("Frequency of data should be no more than daily.")
+        if bigger_than_day(self.freq) and self._mode is not HolidayTransformMode.days_count:
+            raise ValueError("For binary and category modes frequency of data should be no more than daily.")
 
         cols = df.columns.get_level_values("segment").unique()
         out_column = self._get_column_name()
 
-        if self._mode is HolidayTransformMode.category:
+        if self._mode is HolidayTransformMode.days_count:
+            date_offset = pd.tseries.frequencies.to_offset(self.freq)
+            encoded_matrix = np.empty(0)
+            for dt in df.index:
+                start_date, end_date = define_period(date_offset, pd.Timestamp(dt), self.freq)
+                date_range = pd.date_range(start=start_date, end=end_date, freq="d")
+                count_holidays = sum(1 for d in date_range if d in self.holidays)
+                holidays_freq = count_holidays / date_range.size
+                encoded_matrix = np.append(encoded_matrix, holidays_freq)
+        elif self._mode is HolidayTransformMode.category:
             encoded_matrix = np.array(
                 [self.holidays[x] if x in self.holidays else self._no_holiday_name for x in df.index]
             )
-        else:
+        elif self._mode is HolidayTransformMode.binary:
             encoded_matrix = np.array([int(x in self.holidays) for x in df.index])
+        else:
+            raise AssertionError("HolidayTransform supports only ``days_count``, ``binary`` and ``categorical`` mode ")
         encoded_matrix = encoded_matrix.reshape(-1, 1).repeat(len(cols), axis=1)
         encoded_df = pd.DataFrame(
             encoded_matrix,
             columns=pd.MultiIndex.from_product([cols, [out_column]], names=("segment", "feature")),
             index=df.index,
         )
-        encoded_df = encoded_df.astype("category")
-
+        if self._mode is not HolidayTransformMode.days_count:
+            encoded_df = encoded_df.astype("category")
```

Comment:
Why is it "category" if it is a float variable?

Reply:
if it is binary or category mode it's not a float variable